File: src/main/java/com/vspk/bookrest/dto/UserDto.java
package com.vspk.bookrest.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.sun.istack.NotNull;
import com.vspk.bookrest.domain.User;
import lombok.Data;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class UserDto {
    private String username;
    private String firstName;
    private String lastName;
    private String email;

    public User toUser(){
        User user = new User();
        user.setUsername(username);
        user.setFirstName(firstName);
        user.setLastName(lastName);
        user.setEmail(email);

        return user;
    }

    public static UserDto fromUser(User user) {
        UserDto userDto = new UserDto();
        userDto.setUsername(user.getUsername());
        userDto.setFirstName(user.getFirstName());
        userDto.setLastName(user.getLastName());
        userDto.setEmail(user.getEmail());

        return userDto;
    }
}

File: src/main/java/com/vspk/bookrest/dto/UserDto.kt
package com.vspk.bookrest.dto

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.vspk.bookrest.domain.User
import lombok.Data

@JsonIgnoreProperties(ignoreUnknown = true)
data class UserDto(
    private val username: String,
    private val firstName: String,
    private val lastName: String,
    private val email: String
)